src/Microsoft.AspNetCore.Build/targets/Initialize.targets

```diff
@@ -0,0 +1,127 @@
+        <AssemblyFileNumber Condition="'$(AssemblyFileNumber)' == ''">$(_DayBasedVersionNumber)</AssemblyFileNumber>
+
+        <DetectedBuildHost Condition="'$(TEAMCITY_VERSION)' != ''">TeamCity</DetectedBuildHost>
+        <DetectedBuildHost Condition="'$(TRAVIS)' != '' And '$(CONTINUOUS_INTEGRATION)' != ''">Travis</DetectedBuildHost>
+        <DetectedBuildHost Condition="'$(DetectedBuildHost)' == ''">Local Build</DetectedBuildHost>
+        <BuildHost Condition="'$(BuildHost)' == ''">$(DetectedBuildHost)</BuildHost>
+    </PropertyGroup>
+    <PropertyGroup>
+        <!-- Properties derived from the above, not overridable -->
+        <AssemblyFileVersion>$(BuildVersion).$(AssemblyFileNumber)</AssemblyFileVersion>
+        <BuildVersionSuffix>$(BuildQuality)-$(BuildNumber)</BuildVersionSuffix>
+        <BuildSemanticVersion>$(BuildVersion)-$(BuildVersionSuffix)</BuildSemanticVersion>
+
+        <SourceDir>$([System.IO.Path]::Combine($(RepositoryDir), "src"))</SourceDir>
+        <TestDir>$([System.IO.Path]::Combine($(RepositoryDir), "test"))</TestDir>
+        <ToolsDir>$([System.IO.Path]::Combine($(RepositoryDir), "tools"))</ToolsDir>
+        <SamplesDir>$([System.IO.Path]::Combine($(RepositoryDir), "samples"))</SamplesDir>
+        <ArtifactsDir>$([System.IO.Path]::Combine($(RepositoryDir), "artifacts"))</ArtifactsDir>
+        <BuildDir>$([System.IO.Path]::Combine($(ArtifactsDir), "build"))</BuildDir>
+        <PackagesDir>$([System.IO.Path]::Combine($(ArtifactsDir), "packages"))</PackagesDir>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <!-- Locate things to be restored -->
+        <NpmDirs Include="$(RepositoryDir)/**/package.json" />
+        <BowerDirs Include="$(RepositoryDir)/**/bower.json" />
+        <GruntDirs Include="$(RepositoryDir)/**/gruntfile.js" />
+    </ItemGroup>
+
+    <ItemGroup>
+        <UnresolvedProjects Include="$(SourceDir)/**/project.json" />
+        <UnresolvedProjects Include="$(TestDir)/**/project.json" />
+        <UnresolvedProjects Include="$(SamplesDir)/**/project.json" />
+    </ItemGroup>
+
+    <Target Name="CollectCommitHash">
+        <PropertyGroup>
+            <_TempFile>$([System.IO.Path]::GetTempFileName())</_TempFile>
+        </PropertyGroup>
+        <Exec Command="git rev-parse HEAD > $(_TempFile)" WorkingDirectory="$(RepositoryDir)" />
+        <ReadLinesFromFile File="$(_TempFile)">
+            <Output TaskParameter="Lines" PropertyName="CommitHash" />
+        </ReadLinesFromFile>
+        <Delete Files="$(_TempFile)" />
+    </Target>
+
+    <Target Name="DetectOSPlatform">
+        <DetectOSPlatform>
+            <Output TaskParameter="PlatformName" PropertyName="BuildPlatformName" />
+        </DetectOSPlatform>
+        <PropertyGroup>
+            <NuGetPath Condition="'$(BuildPlatformName)' == 'Windows'">$(MSBuildBinPath)\CoreRun.exe $(MSBuildBinPath)\NuGet.CommandLine.XPlat.dll</NuGetPath>
```

Comment:
Would 
`<NuGetPath>$(MSBuildBinPath)\corerun $(MSBuildBinPath)\NuGet.CommandLine.XPlat.dll</NuGetPath>`

work all the time?